src/ProxyManager/ProxyGenerator/NullObject/MethodGenerator/NullObjectMethodInterceptor.php

```diff
@@ -18,9 +19,12 @@ class NullObjectMethodInterceptor extends MethodGenerator
      */
     public static function generateMethod(MethodReflection $originalMethod): self
     {
-        $method = static::fromReflectionWithoutBodyAndDocBlock($originalMethod);
+        $method             = static::fromReflectionWithoutBodyAndDocBlock($originalMethod);
+        $originalReturnType = $originalMethod->getReturnType();
 
-        if ($originalMethod->returnsReference()) {
+        if ($originalReturnType instanceof ReflectionNamedType && $originalReturnType->getName() === 'never') {
+            $method->setBody('throw new \Exception();');
```

Comment:
Is this really needed? without anything in the body, php already throws:
`TypeError: [...]: never-returning function must not implicitly return`

Reply:
Mh, probably is not needed. Don't rembember why I added it honestly, I'll remove that